File: src/com/jda/functionalprograms/LeapYear.java
package com.jda.functionalprograms;

import com.jda.utility.Calendar;
import com.jda.utility.Printer;
import com.jda.utility.Reader;

public
class LeapYear {

  public static
  void main(String[] args) {
    Reader reader = new Reader();
    Printer.print("Enter the year : ");
    int year = reader.readInteger();
    while (year < 1000 || year > 9999) {
      Printer.print("Year must be 4 digit.\nEnter the year: ");
      year = reader.readInteger();
    }
    if (Calendar.isLeapYear(year)) { Printer.printLine("It is a leap year"); }
    else {
      Printer.printLine("It is not a leap year");
    }
  }
}

File: src/com/jda/functionalprograms/LeapYear.kt
package com.jda.functionalprograms

import com.jda.utility.Utility

object LeapYear {

    @JvmStatic
    fun main(args: Array<String>) {
        val reader = Utility().Reader()
        Utility.print("Enter the year : ")
        var year = reader.readInteger()
        while (year < 1000 || year > 9999) {
            Utility.print("Year must be 4 digit.\nEnter the year: ")
            year = reader.readInteger()
        }
        if (Utility.isLeapYear(year)) {
            Utility.printLine("It is a leap year")
        } else {
            Utility.printLine("It is not a leap year")
        }
    }
}
